backend/app/store/search/bleve.go

```diff
@@ -0,0 +1,242 @@
+	langEs "github.com/blevesearch/bleve/v2/analysis/lang/es"
+	langFi "github.com/blevesearch/bleve/v2/analysis/lang/fi"
+	langFr "github.com/blevesearch/bleve/v2/analysis/lang/fr"
+	langIt "github.com/blevesearch/bleve/v2/analysis/lang/it"
+	langRu "github.com/blevesearch/bleve/v2/analysis/lang/ru"
+
+	"github.com/blevesearch/bleve/v2/analysis/token/lowercase"
+	"github.com/blevesearch/bleve/v2/mapping"
+	"github.com/umputun/remark42/backend/app/store"
+)
+
+var analyzerNameMapping = map[string]string{
+	"standard": analyzerStandard.Name,
+	"ar":       langAr.AnalyzerName,
+	"de":       langDe.AnalyzerName,
+	"en":       langEn.AnalyzerName,
+	"es":       langEs.AnalyzerName,
+	"fi":       langFi.AnalyzerName,
+	"fr":       langFr.AnalyzerName,
+	"it":       langIt.AnalyzerName,
+	"ru":       langRu.AnalyzerName,
+}
+
+type bleveEngine struct {
+	index     bleve.Index
+	indexPath string
+}
+
+// Index adds set of comments to the index
+func (b *bleveEngine) Index(comments []store.Comment) error {
+	batch := b.index.NewBatch()
+	for _, comment := range comments {
+		key := DocumentKey{Locator: comment.Locator, ID: comment.ID}
+		keyBytes, err := json.Marshal(key)
+		if err != nil {
+			return fmt.Errorf("can't marshal key %+v: %w", key, err)
+		}
+
+		doc := newBleveDoc(comment)
+		err = batch.Index(string(keyBytes), doc)
+		if err != nil {
+			return fmt.Errorf("can't add to indexing batch: %w", err)
+		}
+	}
+	err := b.index.Batch(batch)
+	if err != nil {
+		return fmt.Errorf("index error: %w", err)
+	}
+
+	return nil
+}
+
+// Search performs search request
+func (b *bleveEngine) Search(req *Request) (*Result, error) {
+	bQuery := bleve.NewQueryStringQuery(req.Query)
+	bReq := bleve.NewSearchRequestOptions(bQuery, req.Limit, req.Offset, false)
+
+	switch {
+	case req.SortBy == "":
+		bReq.SortBy([]string{"-time"})
+	case req.SortBy == "time" || req.SortBy == "-time" || req.SortBy == "+time":
+		bReq.SortBy([]string{req.SortBy})
+	default:
+		return nil, fmt.Errorf("unknown sort field %s", req.SortBy)
+	}
+
+	searchRes, err := b.index.Search(bReq)
+	if err != nil {
+		return nil, fmt.Errorf("search error: %w", err)
+	}
+
+	log.Printf("[INFO] found %d documents for query %q in %s",
+		searchRes.Total, req.Query, searchRes.Took.String())
+
+	result := convertResultRepr(searchRes)
+	return &result, nil
+}
+
+// Size is the number of documents in the index
+func (b *bleveEngine) Size() (uint64, error) {
+	return b.index.DocCount()
+}
+
+// Close closes index
+func (b *bleveEngine) Close() error {
+	return b.index.Close()
+}
+
+func newBleveEngine(indexPath, analyzer string) (*bleveEngine, error) {
+	indexExists, err := checkIndexPath(indexPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var index bleve.Index
+	if !indexExists {
+		log.Printf("[INFO] creating new search index %s", indexPath)
+
+		bleveAnalyzerName, has := analyzerNameMapping[analyzer]
+		if !has {
+			return nil, fmt.Errorf("unknown analyzer %s", analyzer)
```

Comment:
Please cover this case with test.

Reply:
Added in separate test in search module https://github.com/umputun/remark42/pull/757/commits/3e4515b0cbecc0927cfc3a47defbee1f3dd3675e#diff-f933ab083e2ed57c319100cfb2bc25f3fc8493387c6b2fd5531600d3150870aaR261